assets/shared/helpers/player.js

```diff
@@ -0,0 +1,557 @@
+		 */
+		initializePlayer: ( element ) =>
+			new Promise( ( resolve ) => {
+				// eslint-disable-next-line @wordpress/no-global-event-listener -- Not in a React context.
+				window.addEventListener( 'message', ( event ) => {
+					if (
+						event.source !== element.contentWindow ||
+						event.data.event !== 'videopress_durationchange' ||
+						! event.data.durationMs
+					) {
+						return;
+					}
+
+					// Set the duration to a dataset in order to have it available for later.
+					element.dataset.duration =
+						parseInt( event.data.durationMs, 10 ) / 1000;
+
+					resolve( element );
+				} );
+			} ),
+
+		/**
+		 * Get the video duration.
+		 *
+		 * @param {HTMLIFrameElement} player The player element.
+		 *
+		 * @return {Promise<number>} The duration of the video in seconds through a promise.
+		 */
+		getDuration: ( player ) => {
+			const duration = player.dataset.duration;
+			if ( ! duration ) {
+				return Promise.reject(
+					new Error( 'Video duration not found' )
+				);
+			}
+
+			return Promise.resolve( parseFloat( player.dataset.duration ) );
+		},
+
+		/**
+		 * Set the video to a current time.
+		 *
+		 * @param {HTMLIFrameElement} player  The player element.
+		 * @param {number}            seconds The video time in seconds to set.
+		 *
+		 * @return {Promise} A promise that resolves if the video was set to a current time successfully.
+		 */
+		setCurrentTime: ( player, seconds ) => {
+			try {
+				player.contentWindow.postMessage(
+					{
+						event: 'videopress_action_set_currenttime',
+						currentTime: seconds,
+					},
+					'*'
+				);
+				return Promise.resolve();
+			} catch ( e ) {
+				return Promise.reject( e );
+			}
+		},
+
+		/**
+		 * Play the video.
+		 *
+		 * @param {HTMLIFrameElement} player The player element.
+		 *
+		 * @return {Promise} A promise that resolves if the video was played successfully.
+		 */
+		play: ( player ) => {
+			try {
+				player.contentWindow.postMessage(
+					{ event: 'videopress_action_play' },
+					'*'
+				);
+				return Promise.resolve();
+			} catch ( e ) {
+				return Promise.reject( e );
+			}
+		},
+
+		/**
+		 * Pause the video.
+		 *
+		 * @param {HTMLIFrameElement} player The player element.
+		 *
+		 * @return {Promise} A promise that resolves if the video was paused successfully.
+		 */
+		pause: ( player ) => {
+			try {
+				player.contentWindow.postMessage(
+					{ event: 'videopress_action_pause' },
+					'*'
+				);
+				return Promise.resolve();
+			} catch ( e ) {
+				return Promise.reject( e );
+			}
+		},
+
+		/**
+		 * Add an event listener to the player.
+		 *
+		 * @param {HTMLIFrameElement} player    The player element.
+		 * @param {string}            eventName Event name (currently only `timeupdate` is supported)
+		 * @param {Function}          callback  Listener callback.
+		 *
+		 * @return {Function} The function to unsubscribe the event.
+		 */
+		on: ( player, eventName, callback ) => {
```

Comment:
Similar to the above, this seems kinda like it supports any event, but implicitly only supports `timeupdate`. Maybe we should check for the event name?

Reply:
https://github.com/Automattic/sensei/pull/5364#discussion_r929259028